add explorer docs.md (#190)
axonweb3/axon-docs: docs/devops/explorer.md

import useBaseUrl from "@docusaurus/useBaseUrl";

# Explorer
Axon uses [BlockScan](https://github.com/Magickbase/blockscan), an explorer tailored for Ethereum-compatible chains. By deploying BlockScan, you can access data from Axon-based chains and monitor their status.

## Deployment

Docker Compose is the current deployment method.

### Docker Compose

1. Clone BlockScan repository to your target machine:

``` shell
git clone https://github.com/Magickbase/blockscan.git
cd blockscan
```

> 2. Edit the config file located at `- blockscan/dev.env`. Make the following adjustments:

Reviewer: `dev.env` is not mounted in docker-compose file by default.
@Simon-Tl

Author: The reason I want to mount by default is
For developers to connect other axon nodes to explorer, more convenient modification

Reviewer: I think the explanation is needed.

Or,
As a new developer to Axon, I may assume that `dev.env` will be used in `docker-compose` after reading the docs.

So, I think it's better to mount dev.env by default, using cmd similar to:

```bash
docker-compose --env-file ./dev.env ...
```

Author: Yes, I've made dev.env mount by default inside docker-compose, and if dev.env doesn't exist, docker-compose won't work

```shell
ETHEREUM_JSONRPC_VARIANT=geth

> ETHEREUM_JSONRPC_HTTP_URL=https://axon1.layerview.io/

Author: Because it was not certain that axon and explorer would be deployed together, a useless field was filled in for modification

> ETHEREUM_JSONRPC_HTTP_URL=https://axon1.layerview.io/

Reviewer: Please a comment in the docs to tell the developer/doc-reader remembering to modify this config before `docker-compose up`.

# Http address of axon rpc
ETHEREUM_JSONRPC_TRACE_URL=https://axon1.layerview.io/
# Http address of axon rpc
DATABASE_URL=postgresql://postgres:postgres123@db:5432/blockscan?ssl=false

> # DATABASE_URL=postgresql://postgres:postgres@127.0.0.1:5432/blockscan?ssl=false

Reviewer: Why include 2 lines of db config in the docs?

Author: If the developer has already deployed postgres, I think it would be easier to leave a line of comments

Reviewer: yes, you are right, just add your comment in the docs please.

```

3. Execute one-click deployment:

```shell
cd ./bolockscan
sudo docekr-compose up -d
```

4. Access the explorer at:

```shell

> http://xxxx:4020

Reviewer: ```suggestion
http://127.0.0.1:4020
```

import useBaseUrl from "@docusaurus/useBaseUrl";

# Explorer
Axon uses [BlockScan](https://github.com/Magickbase/blockscan), an explorer tailored for Ethereum-compatible chains. By deploying BlockScan, you can access data from Axon-based chains and monitor their status.

## Deployment

Docker Compose is the current deployment method.

### Docker Compose

1. Clone BlockScan repository to your target machine:

``` shell

> git clone https://github.com/Magickbase/blockscan.git

Reviewer: ```suggestion
git clone --depth=1 https://github.com/Magickbase/blockscan.git
```

cd blockscan
```

2. Edit the config file located at `- blockscan/dev.env`. Make the following adjustments:
   - The purpose of the dev.env file is to enable explorer to connect to your locally existing components, but if the axon node and explorer are in the same machine, you can also deploy directly using our docker-compose without any problems

```shell
ETHEREUM_JSONRPC_VARIANT=geth

> # Http address of axon rpc,The default port is port 8000 of the host 

Reviewer: ```suggestion
# Http address of Axon RPC, defaults to port 8000 of the host
```

# Please change to your own axon rpc address port
ETHEREUM_JSONRPC_HTTP_URL=http://host.docker.internal:8000

> # Http address of axon rpc,The default port is port 8000 of the host 

Reviewer: ```suggestion
# Http address of axon rpc,The default port is port 8000 of the host
```

# Please change to your own axon rpc address port
ETHEREUM_JSONRPC_TRACE_URL=http://host.docker.internal:8000
# The default is the postgresql address deployed in docker-compose
# If you have your own postgresql, change it
DATABASE_URL=postgresql://postgres:postgres123@db:5432/blockscan?ssl=false

> 

Reviewer: ```suggestion
```

```

3. Execute one-click deployment:

```shell
cd ./bolockscan

> sudo docekr-compose up -d

Reviewer: ```suggestion
docker-compose up -d --env-file ./dev.env
```
1. typo: docekr -> docker
2. I haven't tested this cmd. Please test it before using it.

Author: ok,no problem

```

4. Access the explorer at:

```shell

> http://127.0.0.1:4020
> #  Replace this address with the actual URL where your BlockScan explorer is deployed

Reviewer: ```suggestion
# Replace this address with the actual URL where your BlockScan explorer is deployed
http://127.0.0.1:4020
```

```

> <img alt="Untitled" src={useBaseUrl("Axon Explorer interface.png")}  width="80%"/>

Reviewer: Let's update the screenshot to the new UI of Blockscan/Blockscout?

Author: yeap

You can deploy explorer to query information on the axon chain and observe the status of the axon chain

### Deployment

#### More diverse deployment methods, gradually developed

#### Docker-compose

#####  1. Clone blockscan to target machine

``` shell
git clone https://github.com/Magickbase/blockscan.git
cd blockscan
```

##### 2. Edit the config config

- blockscan/dev.env

```shell
ETHEREUM_JSONRPC_VARIANT=geth

ETHEREUM_JSONRPC_HTTP_URL=https://axon1.layerview.io/
# Http address of axon rpc
ETHEREUM_JSONRPC_TRACE_URL=https://axon1.layerview.io/
# Http address of axon rpc
DATABASE_URL=postgresql://postgres:postgres123@db:5432/blockscan?ssl=false
# DATABASE_URL=postgresql://postgres:postgres@127.0.0.1:5432/blockscan?ssl=false

```

##### 3. Execute one-click deployment

```shell
cd ./bolockscan
sudo docekr-compose up -d
```

##### 4. Access explorer

```shell
http://xxxx:4020
```

<img alt="Untitled" src={useBaseUrl("Axon Explorer interface.png")}  width="80%"/>

> 
> 
> 
> 
> 
> 🛠 This page is a work in progress. 🚧

Reviewer: ```suggestion
```
